src/snake_game/utils.cljs

```diff
@@ -1,74 +0,0 @@
-(ns snake-game.utils
-  (:require [reagent.core :as reagent :refer [atom]]))
-
-(def key-code->move
-  "Mapping from integer key code to direction vector corresponding to that key"
-  {38 [0 -1]
-   40 [0 1]
-   39 [1 0]
-   37 [-1 0]})
-
-(defn rand-free-position
-  "Fuction takes snake and board-size as arguments.,
-  and returns random position not colliding wit snake body"
-  [snake [x y]]
-  (let [snake-positions (into #{} (:body snake))
-        board-positions (for [x-pos (range x)
-                              y-pos (range y)]
-                          [x-pos y-pos])]
-    (when-let [free-positions (seq (remove snake-positions board-positions))]
-      (rand-nth free-positions))))
-
-(defn collisions
-  "Returns true if snake collision with board edges or itself (snake body) is detected"
-  [{:keys [snake board] :as db}]
```

Comment:
Maybe it would be more clear to input exactly what you need (snake and board) for the collision instead of passing the whole `db` here.